Src/LiquidProjections.PollingEventStore/PollingEventStoreAdapter.cs

```diff
@@ -134,276 +210,190 @@ private Page TryGetNextPageFromCache(long precedingCheckpoint, string subscripti
 
                 throw;
             }
-            
+
             if (transactions.Count > 0)
             {
-#if LIQUIDPROJECTIONS_DIAGNOSTICS
-                logger(() =>
-                    $"Loader for subscription {subscriptionId ?? "without ID"} has loaded {transactions.Count} transactions " +
-                    $"from checkpoint {transactions.First().Checkpoint} to checkpoint {transactions.Last().Checkpoint}.");
-#endif
+                logDebug(() =>
+                    $"Loaded from Event Store: (subscription: {subscriptionId}, precedingCheckpoint: {precedingCheckpoint}, transactions: {transactions.Count}, range: {transactions.First().Checkpoint}-{transactions.Last().Checkpoint}.");
 
                 if (transactions.First().Checkpoint <= precedingCheckpoint)
                 {
                     throw new InvalidOperationException(
                         $"The event store returned a transaction with checkpoint {transactions.First().Checkpoint} that is supposed to be higher than the requested {precedingCheckpoint}");
                 }
-
-                if (cachedTransactionsByPrecedingCheckpoint != null)
-                {
-                    /* Add to cache in reverse order to prevent other projectors
-                        from requesting already loaded transactions which are not added to cache yet. */
-                    for (int index = transactions.Count - 1; index > 0; index--)
-                    {
-                        cachedTransactionsByPrecedingCheckpoint.Set(transactions[index - 1].Checkpoint, transactions[index]);
-                    }
-
-                    cachedTransactionsByPrecedingCheckpoint.Set(precedingCheckpoint, transactions[0]);
-
-#if LIQUIDPROJECTIONS_DIAGNOSTICS
-                    logger(() =>
-                        $"Loader for subscription {subscriptionId ?? "without ID"} has cached {transactions.Count} transactions " +
-                        $"from checkpoint {transactions.First().Checkpoint} to checkpoint {transactions.Last().Checkpoint}.");
-#endif
-                }
             }
             else
             {
-#if LIQUIDPROJECTIONS_DIAGNOSTICS
-                logger(() =>
-                    $"Loader for subscription {subscriptionId} has discovered " +
-                    $"that there are no new transactions yet. Next request for the new transactions will be delayed.");
-#endif
+                logDebug(() =>
+                    $"No Transactions: (subscriptionId: {subscriptionId}, precedingCheckpoint: {precedingCheckpoint})");
             }
 
             return new Page(precedingCheckpoint, transactions);
         }
 
+        private void CachePage(Page page, string subscriptionId)
+        {
+            IReadOnlyList<Transaction> transactions = page.Transactions;
+
+            if (cachedTransactionsByPrecedingCheckpoint != null)
+            {
+                /* Add to cache in reverse order to prevent other projectors

```

Comment:
😕 It took me quite a while before I realized that this would prevent other projectors to request already loaded transactions because the request stays on the queue until this method is done and that projectors will first check if there is already a request on the queue and only then check the cache.

Reply:
#16